File: cluster-cli/src/main/scala/org/midonet/cluster/cli/commands/objects/Network.scala
/*
 * Copyright 2015 Midokura SARL
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package org.midonet.cluster.cli.commands.objects

import java.util.UUID

import org.midonet.cluster.data.{ZoomObject, ZoomField}
import org.midonet.cluster.util.UUIDUtil.{Converter => UUIDConverter}

/** A topology network. */
class Network extends ZoomObject {

    @ZoomField(name = "id", converter = classOf[UUIDConverter])
    var id: UUID = _
    @ZoomField(name = "name")
    var name: String = _
    @ZoomField(name = "tenant_id")
    var tenandId: String = _
    @ZoomField(name = "admin_state_up")
    var adminStateUp: Boolean = _

    @ZoomField(name = "tunnel_key")
    var tunnelKey: Long = _
    @ZoomField(name = "inbound_filter_id", converter = classOf[UUIDConverter])
    var inboundFilter: UUID = _
    @ZoomField(name = "outbound_filter_id", converter = classOf[UUIDConverter])
    var outboundFilter: UUID = _

    @ZoomField(name = "port_ids", converter = classOf[UUIDConverter])
    var portIds: Set[UUID] = _
    @ZoomField(name = "vxlan_port_ids", converter = classOf[UUIDConverter])
    var vxlanPortIds: Set[UUID] = _

    override def toString =
        s"id=$id name=$name tenantId=$tenandId adminStateUp=$adminStateUp " +
        s"tunnelKey=$tunnelKey inboundFilter=$inboundFilter " +
        s"outboundFilter=$outboundFilter portIds=$portIds vxlanPortIds=$vxlanPortIds"

}

File: midonet-cluster/src/main/java/org/midonet/cluster/rest_api/neutron/models/Network.java
/*
 * Copyright 2015 Midokura SARL
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
package org.midonet.cluster.rest_api.neutron.models;

import java.util.UUID;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.google.common.base.MoreObjects;
import com.google.common.base.Objects;

import org.midonet.cluster.data.ZoomClass;
import org.midonet.cluster.data.ZoomField;
import org.midonet.cluster.data.ZoomObject;
import org.midonet.cluster.models.Neutron;

@ZoomClass(clazz = Neutron.NeutronNetwork.class)
public class Network extends ZoomObject {

    public Network() {}

    public Network(UUID id, String tenantId, String name, boolean external) {
        this.id = id;
        this.name = name;
        this.tenantId = tenantId;
        this.shared = false;
        this.adminStateUp = true;
        this.external = external;
    }

    @ZoomField(name = "id")
    public UUID id;

    @ZoomField(name = "name")
    public String name;

    @ZoomField(name = "status")
    public String status;

    @ZoomField(name = "shared")
    public boolean shared;

    @JsonProperty("tenant_id")
    @ZoomField(name = "tenant_id")
    public String tenantId;

    @JsonProperty("admin_state_up")
    @ZoomField(name = "admin_state_up")
    public boolean adminStateUp;

    @JsonProperty("router:external")
    @ZoomField(name = "external")
    public boolean external;

    @JsonProperty("provider:network_type")
    @ZoomField(name = "network_type")
    public NetworkType networkType;

    @JsonProperty("qos_policy_id")
    @ZoomField(name = "qos_policy_id")
    public UUID qosPolicyId;

    @Override
    public boolean equals(Object obj) {
        if (obj == this) return true;
        if (!(obj instanceof Network)) return false;

        final Network other = (Network) obj;
        return Objects.equal(id, other.id)
                && Objects.equal(name, other.name)
                && Objects.equal(status, other.status)
                && Objects.equal(shared, other.shared)
                && Objects.equal(tenantId, other.tenantId)
                && Objects.equal(adminStateUp, other.adminStateUp)
                && Objects.equal(external, other.external)
                && Objects.equal(networkType, other.networkType)
                && Objects.equal(qosPolicyId, other.qosPolicyId);
    }

    @Override
    public int hashCode() {
        return Objects.hashCode(id, name, status, shared, tenantId,
                adminStateUp, external, networkType, qosPolicyId);
    }

    @Override
    public String toString() {

        return MoreObjects.toStringHelper(this)
                .add("id", id)
                .add("name", name)
                .add("status", status)
                .add("shared", shared)
                .add("tenantId", tenantId)
                .add("adminStateUp", adminStateUp)
                .add("external", external)
                .add("networkType", networkType)
                .add("qosPolicyId", qosPolicyId)
                .toString();
    }
}
